### PR TITLE
[REF] remove instances of pass-by-reference where no change takes place

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Further removal of an obsolete code pattern (after grepping)

Before
----------------------------------------
$params passed by reference

After
----------------------------------------
$params not passed by reference

Technical Details
----------------------------------------
I did a grep on these and the passed params are not altered - down with php4

Comments
----------------------------------------
@colemanw

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

File: CRM/Core/BAO/SchemaHandler.php [hunk 1]
```diff
@@ -295,7 +294,7 @@ public static function buildForeignKeySQL($params, $separator, $prefix, $tableNa
    *
    * @return bool
    */
-  public static function alterFieldSQL(&$params, $indexExist = FALSE, $triggerRebuild = TRUE) {
+  public static function alterFieldSQL($params, $indexExist = FALSE, $triggerRebuild = TRUE) {
 
     // lets suppress the required flag, since that can cause sql issue
     $params['required'] = FALSE;
```
Comment (monishdeb) [inline CRM/Core/BAO/SchemaHandler.php]: @eileenmcnaughton do we need to carry forward this assignment?

Comment (eileenmcnaughton) [reply CRM/Core/BAO/SchemaHandler.php]: The function that calls it (the only one) doesn't make use of the $params afterwards

Comment (monishdeb) [reply CRM/Core/BAO/SchemaHandler.php]: Yup make sense.

**Comment by monishdeb** [comment]
Its a straightforward change and makes sense. Merging it now.

**Pull request merged**